internal/autometrics/ctx.go

```diff
@@ -28,13 +29,106 @@ type GeneratorContext struct {
 	//
 	// This can be set in the command for the generator or through the environment.
 	DisableDocGeneration bool
+	// ImportMap maps the alias to import in the current file, to canonical names associated with that name.
+	ImportsMap map[string]string
+}
+
+// This is almost a carbon copy of the autometrics.Context structure, except that
+// non-literal types are transcribed to strings to make it possible to reason about
+// the runtime context within generator logic.
+// Also, all the information that only gets filled at runtime is simply ignored, as
+// only statically-known information is useful at this point.
+type RuntimeCtxInfo struct {
+	// Name of the variable to use as context.Context when building the autometrics.Context.
+	// The string will be empty iff 'nil' must be used as autometrics.NewContext() argument.
+	ContextVariableName string
+	// Verbatim code to use to fetch the TraceID.
+	// For example, if the instrumented function is detected to use Gin, like
+	// `func ginHandler(ginVarName *gin.Context)`
+	// then the code in that variable should be something like
+	// `"autometrics.DecodeString(ginVarName.GetString(\"TraceID\"))"`
+	// This getter should return ([]byte, error) to allow PreInstrument to
```

Comment:
This comment is wrong, the getter should return `[]byte` directly, as there's no way to assign multiple return values to a single variable in Go. Instead of checking for `error`, implementation of `DecodeString` is supposed to transmit `nil` on error, which then requires implementers of `With(Trace|Span)ID` to nil-check their arguments.